numpy/lib/function_base.py

```diff
@@ -847,34 +859,33 @@ def piecewise(x, condlist, funclist, *args, **kw):
 
     Parameters
     ----------
-    x : ndarray
-        The input domain.
+    xi : tuple
+        input arguments to the functions in funclist, i.e., (x0, x1,...., xn)
     condlist : list of bool arrays
         Each boolean array corresponds to a function in `funclist`.  Wherever
-        `condlist[i]` is True, `funclist[i](x)` is used as the output value.
-
-        Each boolean array in `condlist` selects a piece of `x`,
-        and should therefore be of the same shape as `x`.
+        `condlist[i]` is True, `funclist[i](x0,x1,...,xn)` is used as the
+        output value. Each boolean array in `condlist` selects a piece of `xi`,
+        and should therefore be of the same shape as `xi`.
 
         The length of `condlist` must correspond to that of `funclist`.
         If one extra function is given, i.e. if
         ``len(funclist) - len(condlist) == 1``, then that extra function
         is the default value, used wherever all conditions are false.
-    funclist : list of callables, f(x,*args,**kw), or scalars
+    funclist : list of callables, f(*(xi + args), **kw), or scalars
```

Comment:
```suggestion
    funclist : list of callables, f(*xi, *args, **kw), or scalars
```
is also legal and probably clearer

Reply:
xi may be slightly misleading since these funcs run on indexed array as such which will be 1d and may not be all elements of xi.